File: back/app/routers/analytics.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas import AnalyticsOut, ProgressOut
from app.services.analytics_service import compute_analytics, compute_progress

router = APIRouter(tags=["analytics"])


@router.get("/progress/{account_id}", response_model=ProgressOut)
def progress(account_id: str, db: Session = Depends(get_db)) -> ProgressOut:
    data = compute_progress(db, account_id)
    return ProgressOut(
        completed_sessions=int(data["completed_sessions"]),
        completion_rate=float(data["completion_rate"]),
        average_rpe=float(data["average_rpe"]),
        readiness_average=float(data["readiness_average"]),
    )


@router.get("/analytics/{account_id}", response_model=AnalyticsOut)
def analytics(account_id: str, db: Session = Depends(get_db)) -> AnalyticsOut:
    data = compute_analytics(db, account_id)
    return AnalyticsOut(**data)

File: front/athli-app/lib/analytics.ts
import { apiFetch } from './api';

export function getProgress(accountId: string) {
  return apiFetch<{ completed_sessions: number; completion_rate: number; average_rpe: number; readiness_average: number }>(
    `/progress/${accountId}`,
  );
}

export function getAnalytics(accountId: string) {
  return apiFetch<{ weekly_sessions_done: number; weekly_sessions_planned: number; injury_risk_flag: boolean; next_session_intensity: number | null }>(
    `/analytics/${accountId}`,
  );
}
